### ISSUE TITLE
Styled text not merged using backspace

### ISSUE THREAD

**Issue by cameronvoell** [wordpress-mobile/gutenberg-mobile#1873]
**Describe the bug**
This issue is a more specific case of https://github.com/wordpress-mobile/gutenberg-mobile/issues/1865 that only occurs if you style text at the front of a paragraph block, and then try and delete some of the styled text, continuing to hit backspace, until it should merge. 

**To Reproduce**
1. Add a pararaph block to an article and add some text to it
2. Click in the middle of the text and hit enter to move the right half of the text to a new block
3. Hit backspace and watch the new block merge back correctly with the previous block
4. Hit enter from the middle of the text again to create a new block
5. This time select a continuous group of character from the front of the block, and style them as italic or bold.
6. Delete one or more characters from the beginning of the new text block, leaving some of the styled text to the right of your cursor.
7. Continue to hit the backspace key => note that the block is no longer merging with the previous block like it was in step 3

**Expected behavior**
Backspace at the beginning of a paragraph block should always merge it with a previous paragraph block.

**Screenshots**
![styled-text-no-merge-on-delete](https://user-images.githubusercontent.com/1103838/74063401-34a6ce80-49a5-11ea-823d-469a4aecb657.gif)

**Smartphone (please complete the following information):**
 - Pixel 3 Emulator, Android 9

**Comment by mchowning** [comment]
I think I ran into another variation on this bug. I found that after splitting blocks if you added an entire word (by swiping), then deleted that word and tried to merge the blocks, they would not merge. If you just tap letters after splitting the blocks though, they will merge.

![backspace_merge mp4](https://user-images.githubusercontent.com/4656348/78295475-35b63100-74fa-11ea-94b0-217a080bfa8a.gif)
